rmw_fastrtps_shared_cpp/src/rmw_response.cpp

```diff
@@ -104,6 +104,22 @@ __rmw_send_response(
   wparams.related_sample_identity().sequence_number().low =
     (int32_t)(request_header->sequence_number & 0xFFFFFFFF);
 
+  // According to the list of possible entity kinds in section 9.3.1.2 of RTPS
+  // readers will have this bit on, while writers will not. We use this to know
+  // if the related guid is the request writer or the response reader.
+  constexpr uint8_t entity_id_is_reader_bit = 0x04;
+  const eprosima::fastrtps::rtps::GUID_t & related_guid =
+    wparams.related_sample_identity().writer_guid();
+  if((related_guid.entityId.value[3] & entity_id_is_reader_bit) != 0)
+  {
```

Comment:
I think this will cause a linter error
```suggestion
  if ((related_guid.entityId.value[3] & entity_id_is_reader_bit) != 0) {
```

Reply:
Fixed on 13c540b